### ISSUE TITLE
FastZip does not allow Extended Path Filter to be specified

### ISSUE THREAD

**Issue by atcobb01** [icsharpcode/SharpZipLib#405]
Version of SharpZipLib: 1.2.0

trying to use Extended Path Filter, but don't see any examples.  FastZip uses strings for Name and Directory filters, but I need file time as a criteria.  I don't see how to use the ExtendedPathFilter class when creating a zip file using FastZip or any other method.  Can you help?

**Comment by piksel** [comment]
No, there does not seem to be any way to provide custom filters for `FastZip`. I'm not sure about what the purpose of `ExtendedPathFilter` is. It looks like it's meant to be used in `FastZip`, perhaps with an overloaded constructor. That should be a pretty small PR. 

Otherwise, your use case could be achieved fairly easily with `ZipFile` like this:
```cs
using (var zf = new ZipFile(zipFileName)) {
    zf.BeginUpdate();
    var yesterday = DateTime.UtcNow.AddDays(-1);
    var files = Directory.GetFiles(sourceDir)
        .Where(f => new FileInfo(f).LastWriteTimeUtc > yesterday);
        
    foreach(var file in files){
        // Provide an entry name as a second parameter 
        // to avoid using the full path to the file
        var entryName = Path.GetFileName(file);
        zf.Add(file, entryName);
    }
    zf.CommitUpdate();
}
```

**Comment by Numpsy** [comment]
Maybe an extra version of ```CreateZip``` that takes the file/directory filters as ```IScanFilter``` rather than strings?

It looks like ```FileSystemScanner``` already has a constructor that can take a pair of IScanFilters on top of the one that uses strings (currently used by FastZip), so maybe that's a straight forward way of passing in alternate filter implementations?

**Comment by Numpsy** [comment]
This one should be fixed by #482

**Comment by piksel** [comment]
This should be fixed by #482.
